cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -17,177 +17,118 @@ limitations under the License.
 package kubeconfig
 
 import (
-	"bytes"
 	"crypto/x509"
 	"fmt"
-	"os"
-	"path/filepath"
 
-	"k8s.io/client-go/tools/clientcmd"
+	"crypto/rsa"
+
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	certutil "k8s.io/client-go/util/cert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 	"k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/pkiutil"
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 )
 
-// BuildConfigProperties holds some simple information about how this phase should build the KubeConfig object
-type BuildConfigProperties struct {
-	CertDir         string
-	ClientName      string
-	Organization    []string
-	APIServer       string
-	Token           string
-	MakeClientCerts bool
+// buildConfigProperties holds some simple information about how to build the KubeConfig object
+type buildConfigProperties struct {
+	ClientName   string
+	Organization []string
+	APIServer    string
+	Token        string
 }
 
-// TODO: Make an integration test for this function that runs after the certificates phase
-// and makes sure that those two phases work well together...
-
-// TODO: Integration test cases:
-// /etc/kubernetes/{admin,kubelet}.conf don't exist => generate kubeconfig files
-// /etc/kubernetes/{admin,kubelet}.conf and certs in /etc/kubernetes/pki exist => don't touch anything as long as everything's valid
-// /etc/kubernetes/{admin,kubelet}.conf exist but the server URL is invalid in those files => error
-// /etc/kubernetes/{admin,kubelet}.conf exist but the CA cert doesn't match what's in the pki dir => error
-// /etc/kubernetes/{admin,kubelet}.conf exist but not certs => certs will be generated and conflict with the kubeconfig files => error
-
-// CreateInitKubeConfigFiles is called from the main init and does the work for the default phase behaviour
-func CreateInitKubeConfigFiles(masterEndpoint, pkiDir, outDir, nodeName string) error {
-
-	// Create a lightweight specification for what the files should look like
-	filesToCreateFromSpec := map[string]BuildConfigProperties{
```

Comment:
I'd propose to just move this map to cmd and consume it there.
Also keep most of the code as-is here in this pkg.
The only thing that _has_ to change is that and the BuildConfigProperties struct organization